app/models/in_person_enrollment.rb

```diff
@@ -12,8 +12,37 @@ class InPersonEnrollment < ApplicationRecord
 
   validate :profile_belongs_to_user
 
+  before_save :set_status_updated_at, if: :will_save_change_to_status?
+
+  # Find enrollments that need a status check via the USPS API
+  def self.needs_usps_status_check check_interval
+    where(status: :pending).
+    and(
+      where(status_check_attempted_at: check_interval).
+      or(where(status_check_attempted_at: nil)),
+    ).
+    order(status_check_attempted_at: :asc)
+  end
+
+  # Does this enrollment need a status check via the USPS API?
+  def needs_usps_status_check? check_interval
+    status == :pending && (
+      status_check_attempted_at.nil? ||
+      check_interval.cover?(status_check_attempted_at)
+    )
+  end
+
+  # Returns the value to use for the USPS enrollment ID
+  def usps_unique_id
+    user_id.to_s
+  end
+
   private
 
+  def set_status_updated_at
```

Comment:
I have a personal style thing against using `set_` in ruby names. But also, do we need this method at all? Could we update the attribute in the job directly?

Reply:
changed to `on_status_updated` in 4f696d143f09125ceebc32db5f2c0165a0be7a89